tensorflow/contrib/ignite/kernels/ignite_dataset_iterator.cc

```diff
@@ -0,0 +1,437 @@
+
+  int32_t row_cnt;
+  TF_RETURN_IF_ERROR(client_->ReadInt(&row_cnt));
+
+  remainder_ = res_len - 25;
+  page_ = std::unique_ptr<uint8_t>(new uint8_t[remainder_]);
+  ptr_ = page_.get();
+
+  int64_t start = std::chrono::duration_cast<std::chrono::milliseconds>(
+                      std::chrono::system_clock::now().time_since_epoch())
+                      .count();
+
+  TF_RETURN_IF_ERROR(client_->ReadData(ptr_, remainder_));
+
+  int64_t stop = std::chrono::duration_cast<std::chrono::milliseconds>(
+                     std::chrono::system_clock::now().time_since_epoch())
+                     .count();
+  ;
+
+  double size_in_mb = 1.0 * remainder_ / 1024 / 1024;
+  double time_in_s = 1.0 * (stop - start) / 1000;
+  LOG(INFO) << "Page size " << size_in_mb << " Mb, time " << time_in_s * 1000
+            << " ms download speed " << size_in_mb / time_in_s << " Mb/sec";
+
+  uint8_t last_page_b;
+  TF_RETURN_IF_ERROR(client_->ReadByte(&last_page_b));
+
+  last_page_ = !last_page_b;
+
+  return Status::OK();
+}
+
+Status IgniteDatasetIterator::LoadNextPage() {
+  TF_RETURN_IF_ERROR(client_->WriteInt(18));  // Message length
+  TF_RETURN_IF_ERROR(
+      client_->WriteShort(load_next_page_opcode));     // Operation code
+  TF_RETURN_IF_ERROR(client_->WriteLong(0));           // Request ID
+  TF_RETURN_IF_ERROR(client_->WriteLong(cursor_id_));  // Cursor ID
+
+  int64_t wait_start = std::chrono::duration_cast<std::chrono::milliseconds>(
+                           std::chrono::system_clock::now().time_since_epoch())
+                           .count();
+
+  int32_t res_len;
+  TF_RETURN_IF_ERROR(client_->ReadInt(&res_len));
+
+  int64_t wait_stop = std::chrono::duration_cast<std::chrono::milliseconds>(
+                          std::chrono::system_clock::now().time_since_epoch())
+                          .count();
+
+  LOG(INFO) << "Load Next Page waited " << (wait_stop - wait_start) << " ms";
+
+  if (res_len < 12)
+    return errors::Internal("Load Next Page Response is corrupted");
+
+  int64_t req_id;
+  TF_RETURN_IF_ERROR(client_->ReadLong(&req_id));
+
+  int32_t status;
+  TF_RETURN_IF_ERROR(client_->ReadInt(&status));
+
+  if (status != 0) {
+    uint8_t err_msg_header;
+    TF_RETURN_IF_ERROR(client_->ReadByte(&err_msg_header));
+
+    if (err_msg_header == string_val) {
+      int32_t err_msg_length;
+      TF_RETURN_IF_ERROR(client_->ReadInt(&err_msg_length));
+
+      uint8_t* err_msg_c = new uint8_t[err_msg_length];
+      TF_RETURN_IF_ERROR(client_->ReadData(err_msg_c, err_msg_length));
+      std::string err_msg((char*)err_msg_c, err_msg_length);
+      delete[] err_msg_c;
+
+      return errors::Internal("Load Next Page Error [status=", status,
+                              ", message=", err_msg, "]");
+    }
+    return errors::Internal("Load Next Page Error [status=", status, "]");
+  }
+
+  int32_t row_cnt;
+  TF_RETURN_IF_ERROR(client_->ReadInt(&row_cnt));
+
+  remainder_ = res_len - 17;
```

Comment:
Where does 17 come from? Use named constants for quantities like this to make the code easier to follow.

Reply:
Replaced by correspondent constant.